### ISSUE TITLE
What did we learn about agile culture, tools, and practices?

### ISSUE THREAD

**Issue by k8forz** [IITIDIDX597/sp_2021_team2#45]
- [x] Kait
- [x] Julian
- [x] Arijit
- [x] Sanhyong

**Comment by arijit1035** [comment]
Design thinking and Agile applied together become strong drivers of innovation. While design thinking can help in understanding a problem and its root cause, Agile tools and processes can help in creating solutions to the problems. Agile tools helps in efficient, effective and deep team collaboration to deliver solutions within a specified timeline. However Agile is not about the speed of work but the team dynamics, where friction among the members is minimal and everyone understands their role in a project. Therefore the most important part before delving into Agile practice is to form a team with proper skills and knowledge (T shaped designers preferred) that can take on the challenges head on.

**Comment by akallish** [comment]
@arijit1035 well said. Agile breaks down the desired value into decomposed epics, stories and tasks that trace back to a goal.

**Comment by arijit1035** [comment]
@akallish Exactly, I feel for us as designers diverging a lot during our thinking becomes a problem sometimes. The goal, epic, stories also helps to stay in track.

**Comment by akallish** [comment]
@arijit1035 that is why agile's emphasis on decomposition and tracability is critical in connecting strategy with tasks.

**Comment by SanhyongPark** [comment]
Agile is about constantly building, measuring, learning, and making iterative improvements. Agile’s view is that changes always improve a project. Changes provide additional value. In the long run, small, incremental improvements to a product are a much more efficient path to success. 

Agile requires well-sized stories to keep team members nimble and keep making progress. If a story is too big when priorities shift you’ll have invested a bunch of time into work that can’t be put into production. If a story is too small, when priorities shift you’ll be left with a bunch of work fragments that are “done” but not useful without more time.

**Comment by k8forz** [comment]
**What I learned about agile culture, tools, and practices:**

Agile is not a specific methodology, but a set of values and principles that give teams a common foundation to make decisions that result in a streamlined workflow and a more relevant, useful outcome for the user. It's not a rigid way of working, but a surprisingly flexible approach that enables the team to complete work in a logical manner. 

For instance, in an Agile culture, the team's practices support Agile values and principles outlined the Agile manifesto. One of the principles in the manifesto is "Simplicity --the art of maximizing the amount of work not done-- is essential." As a team, you fulfill this value by having a traceable logic for your entire work plan - whether it's using the IBM framework of hills, subhills, epics, stories, etc.-- each unit of work has to be directly or indirectly linked to your North Star.

**Comment by walkoutthebox** [comment]
**What I learned about agile culture, tools, and practices:**

What immediately comes to mind is that I learned how the agile methodology fits into the design process, what it's good for and what it's not so good for. Coming into the class, I honestly had no idea what Agile was, only that designers spoke about it as if it was incompatible with the design process. After taking the class, I've come to know that agile in a lot of ways is actually very complimentary to the design process. Design thinking is successful in the problem solving phase of the design of a new product/service. Once a high level solution or goal has been established through the traditional design process, agile is successful in ensuring that teams develop solutions that are in line with the goal. In some ways it is a system of accountability that keeps designers/ agile teams honest, making sure that they develop a product that is in line with what users said they wanted during the testing phase of the design process. 

Agile culture on the other hand is really centered around developing teams with great team dynamics in order to deliver consistently good work. This entails developing a collective understanding of each team members roles and responsibilities in maintaining "good" team dynamics as well as ensuring the team has skills and tools to deliver great work consistently. 

I also learned that agile and agile culture are also very focused on eliminating wasted resources, time, and effort. The goal is to allow teams to focus on what matters most in reaching their project goals. The tools involved, github, zenhub, etc. assist in allowing teams to move through project work with minimal overhead and wasted time.

**Comment by k8forz** [comment]
In conclusion:

- We learned how the agile methodology fits into the design process. 

- Agile is not a specific methodology, but a set of values and principles.

- Agile culture is really centered around developing teams with great team dynamics.

- Agile is about constantly building, measuring, learning, and making iterative improvements.
